src/metabase/events/audit_log.clj

```diff
@@ -1,100 +1,93 @@
    [toucan2.core :as t2]))
 
 (derive ::event :metabase/event)
 
-(defn maybe-prepare-update-event-data
-  "When `:audit-db/previous` is present in the event-data, we return a map with previous and new versions of the
-  objects, _keeping only fields that changed_.
-
-  If `:audit-db/previous` is missing, this is a noop."
-  [event]
-  (if-let [previous (:audit-log/previous event)]
-    (let [new (dissoc event :audit-log/previous)
-          [previous-only new-only _both] (data/diff previous new)
-          updated-keys (distinct (concat (keys previous-only) (keys new-only)))]
-      {:previous (select-keys previous updated-keys)
-       :new (select-keys new updated-keys)})
-    event))
-
 (derive ::card-event ::event)
 (derive :event/card-create ::card-event)
 (derive :event/card-update ::card-event)
 (derive :event/card-delete ::card-event)
 
 (methodical/defmethod events/publish-event! ::card-event
-  [topic {card :object :as _event}]
-  (audit-log/record-event! topic card))
+  [topic event]
+  (audit-log/record-event! topic event))
 
 (derive ::card-read-event :metabase/event)
 (derive :event/card-read ::card-read-event)
 
 (methodical/defmethod events/publish-event! ::card-read-event
-  [topic card]
-  (audit-log/record-event! topic card))
+  [topic event]
+  (audit-log/record-event! topic event))
 
 (derive ::card-query-event ::event)
 (derive :event/card-query ::card-query-event)
 
 (methodical/defmethod events/publish-event! ::card-query-event
-  [topic {:keys [card_id] :as object}]
+  [topic {:keys [user-id card-id] :as object}]
   (let [details (select-keys object [:cached :ignore_cache :context])]
-    (audit-log/record-event! topic details api/*current-user-id* :model/Card card_id)))
+    (audit-log/record-event! topic {:details    details
+                                    :user-id    user-id
+                                    :model      :model/Card
+                                    :model-id   card-id})))
 
 (derive ::dashboard-event ::event)
 (derive :event/dashboard-create ::dashboard-event)
 (derive :event/dashboard-delete ::dashboard-event)
 
 (methodical/defmethod events/publish-event! ::dashboard-event
-  [topic {dashboard :object :as _event}]
-  (audit-log/record-event! topic dashboard))
+  [topic event]
+  (audit-log/record-event! topic event))
 
 (derive ::dashboard-card-event ::event)
 (derive :event/dashboard-add-cards ::dashboard-card-event)
 (derive :event/dashboard-remove-cards ::dashboard-card-event)
 
-;; TODO (crumb)
 (methodical/defmethod events/publish-event! ::dashboard-card-event
-  [topic {:keys [dashcards id]}]
+  [topic {:keys [object dashcards user-id] :as _event}]
   ;; we expect that the object has just a dashboard :id at the top level
   ;; plus a `:dashcards` attribute which is a vector of the cards added/removed
-  (let [details (-> (t2/select-one [:model/Dashboard :description :name] :id id)
+  (let [details (-> (select-keys object [:description :name :id])
```

Comment:
I just noticed we have an N+1 query situation here. (`t2/select` N times rather than something like

```clojure
(methodical/defmethod events/publish-event! ::dashboard-card-event
  [topic {:keys [object dashcards user-id] :as _event}]
  ;; we expect that the object has just a dashboard :id at the top level
  ;; plus a `:dashcards` attribute which is a vector of the cards added/removed
  (let [cards   (when (seq dashcards)
                  (t2/select-fn->fn :id #(select-keys % [:name :description])
                                    :model/Card
                                    :id [:in (map :card_id dashcards)]))
        details (-> (select-keys object [:description :name :id])
                    (assoc :dashcards (for [{:keys [id card_id]} dashcards]
                                        (-> (cards card_id)
                                            (assoc :id id)
                                            (assoc :card_id card_id)))))]
    (audit-log/record-event! topic
                               {:details  details
                                :user-id  user-id
                                :model    :model/Dashboard
                                :model-id (u/id object)})))
```

Reply:
Good call, will update